Assets/MixedRealityToolkit/Providers/UnityInput/UnityTouchDeviceManager.cs

```diff
@@ -34,11 +34,21 @@ public class UnityTouchDeviceManager : BaseInputDeviceManager
 
         private static readonly Dictionary<int, UnityTouchController> ActiveTouches = new Dictionary<int, UnityTouchController>();
 
+        private List<UnityTouchController> touchesToRemove = new List<UnityTouchController>();
+
         /// <inheritdoc />
         public override void Update()
         {
+            // Ensure that touch up and source lost events are at least one frame apart.
```

Comment:
Was this to match the behavior with source detected / input down, or was there an issue when they happen on the same frame?

Reply:
i believe the comment is incorrect... i was trying to recall specifically our rationale here :)

Reply:
i'll update the comment

Reply:
There’s an issue when they happen on the same frame. I believe if you delete this code, then button releases will not work on Unity UI—buttons will stay pressed.

Reply:
David, I’m a bit confused...why do you think this is not correct? Is it because the comment does not explain why we need to ensure touch up and lost are not on same frame?

Reply:
The comment was incorrect. Kurtis' comment is the accurate one.